### PR TITLE
Add EditorPythonBindings as an explicit dependency of the White Box Tool

### PR DESCRIPTION
**Pull request by hultonha**
Fixes https://github.com/o3de/o3de/issues/6424

### CODE REVIEW

**Review by jackalbe — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]
All the `"dependencies"` field is used for in the gem.json is for ProjectManager to indicate which gems the WhiteBox gem depends on in the UI,
This doesn't actually add a real build and load dependency.

This needs to be setup in the `RUNTIME_DEPENDENCIES` for the `WhiteBox.Editor` target in the CMakeLists.txt file

**Comment by hultonha** [comment]
> `RUNTIME_DEPENDENCIES`

Ah okay good to know 👍 I'll update this shortly

**Comment by hultonha** [comment]
@lumberyard-employee-dm is it worth keeping the entry in the 'dependencies' for the Project Manager? I take it that would still be useful?

**Comment by lumberyard-employee-dm** [comment]
> @lumberyard-employee-dm is it worth keeping the entry in the 'dependencies' for the Project Manager? I take it that would still be useful?

Yes Project Manager uses that to notify users what gems are implicitly activated when enabling a specific gem.

File: Gems/WhiteBox/Code/CMakeLists.txt [hunk 1]
```diff
@@ -44,6 +43,9 @@ if(NOT PAL_TRAIT_WHITEBOX_SUPPORTED)
             BUILD_DEPENDENCIES
                 PRIVATE
                     AZ::AzCore
+            RUNTIME_DEPENDENCIES
+                PRIVATE
+                    Gem::EditorPythonBindings.Editor
```
Comment (lumberyard-employee-dm) [inline Gems/WhiteBox/Code/CMakeLists.txt:48]: `RUNTIME_DEPENDENCIES` doesn't have use scope for target propagation.
`PRIVATE` needs to be removed
```suggestion
            RUNTIME_DEPENDENCIES
                Gem::EditorPythonBindings.Editor
```

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by SergeyAMZN — APPROVED** [review]

**Pull request merged**